oshi-core/src/main/java/oshi/driver/unix/openbsd/disk/Disklabel.java

```diff
@@ -0,0 +1,127 @@
+        List<String> cmd = ExecutingCommand.runNative("disklabel -n " + diskName);
+        if (cmd.size() > 0) {
+            String[] split;
+            String line;
+            ListIterator<String> lines = cmd.listIterator();
+            while (lines.hasNext()) {
+                line = lines.next();
+                if (line.startsWith("16 partitions:")) {
+                    break;
+                }
+                split = line.split(":");
+                if (split.length == 2) {
+                    params.put(split[0], split[1].trim());
+                }
+            }
+            // should be just the partition table left
+            HWPartition p;
+            String name;
+            while (lines.hasNext()) {
+                line = lines.next();
+                //16 partitions:
+                //#                size           offset  fstype [fsize bsize   cpg]
+                //  a:          2097152             1024  4.2BSD   2048 16384 12958 # /
+                //  b:         17023368          2098176    swap                    # none
+                //  c:        500118192                0  unused
+                //  d:          8388576         19121568  4.2BSD   2048 16384 12958 # /tmp
+                //  e:         41386752         27510144  4.2BSD   2048 16384 12958 # /var
+                //  f:          4194304         68896896  4.2BSD   2048 16384 12958 # /usr
+                //  g:          2097152         73091200  4.2BSD   2048 16384 12958 # /usr/X11R6
+                //  h:         20971520         75188352  4.2BSD   2048 16384 12958 # /usr/local
+                //  i:              960               64   MSDOS
+                //  j:          4194304         96159872  4.2BSD   2048 16384 12958 # /usr/src
+                //  k:         12582912        100354176  4.2BSD   2048 16384 12958 # /usr/obj
+                //  l:        387166336        112937088  4.2BSD   4096 32768 26062 # /home
+                // Note size is in sectors
+                if (line.startsWith("#") || line.isEmpty()) {
+                    continue;
+                }
+                split = ParseUtil.whitespaces.split(line.trim(), 9);
+                name = split[0].replaceAll(":", "");
+                if (split.length > 5) {
+                    p = new HWPartition(diskName + name, name, split[3], Constants.UNKNOWN,
+                        ParseUtil.parseLongOrDefault(split[1], -1) * ParseUtil
+                            .parseLongOrDefault(params.get("bytes/sector"), 1), -1, -1,
+                        split[split.length - 1]);
+                    partitions.add(p);
+                } else if (split.length == 4) {
+                    p = new HWPartition(diskName + name, name, split[3], Constants.UNKNOWN,
+                        ParseUtil.parseLongOrDefault(split[1], -1) * ParseUtil
+                            .parseLongOrDefault(params.get("bytes/sector"), -1), -1, -1, "");
```

Comment:
One more note: you're using -1 for major and minor. I believe these values are accessible as the filesize of the device.  See `ls -al` output for /dev for the given partition to see what I mean.  I think you can extract it using `stat` -- see the code in `oshi.driver.unix.freebsd.disk.GeomPartList.java`.  If root permissions don't allow that, you can see it in `ls` output.